libpod/container_internal_linux.go

```diff
@@ -193,9 +226,8 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 			return nil, errors.Wrapf(err, "error adding systemd-specific mounts")
 		}
 	}
-
 	// Look up and add groups the user belongs to, if a group wasn't directly specified
-	if !rootless.IsRootless() && !strings.Contains(c.config.User, ":") {
+	if c.config.User != "" && !rootless.IsRootless() {
```

Comment:
I think we still want the strings.Contains thing here, at the very end

Reply:
well I noodled on that.  would you do !strings.Container || != "" ?

Reply:
I think we want the config.User != "" first, the order on the other two probably doesn't matter

Reply:
fixed